priv/perf/apps/load_test/lib/ethereum/nonce_tracker.ex

```diff
@@ -0,0 +1,42 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule LoadTest.Ethereum.NonceTracker do
+  @moduledoc """
+  Nonce tracker for sending ethereum transactions
+  """
+
+  alias ExPlasma.Encoding
+
+  def init() do
+    :ets.new(:nonce_tracker, [:set, :public, :named_table])
```

Comment:
Yeah but it comes with a memory and more expensive reading tradeoffs. As we have at least the same number of reads as writes I am not sure there is a point to optimize that if we don't suffer from performance issues.

Reply:
```
Even though the access of a single object is always guaranteed to be atomic and isolated, each traversal through a table to find the next key is not done with such guarantees. This is often not a problem, but may cause rare subtle "unexpected" effects if a concurrent process inserts objects during a traversal. For example, consider one process doing

ets:new(t, [ordered_set, named_table]),
ets:insert(t, {1}),
ets:insert(t, {2}),
ets:insert(t, {3}),
A concurrent call to ets:first(t), done by another process, may then in rare cases return 2 even though 2 has never existed in the table ordered as the first key. In the same way, a concurrent call to ets:next(t, 1) may return 3 even though 3 never existed in the table ordered directly after 1.

Effects like this are improbable but possible. The probability will further be reduced (if not vanish) if table option write_concurrency is not enabled. This can also only be a potential concern for ordered_set where the traversal order is defined.
```
https://erlang.org/doc/man/ets.html

You're optimising for concurrent access.

Reply:
That might not apply for counters tho... true.